src/alloc.rs

```diff
@@ -1,12 +1,20 @@
-//! Optimized audio signal data structures, used in AudioProcessors
+//! Optimized audio signal data structures, used in `AudioProcessors`
+#![warn(
+    clippy::all,
+    clippy::perf,
+    // clippy::pedantic,
+    // clippy::nursery,
+    // clippy::missing_docs_in_private_items
+)]
 
 use arrayvec::ArrayVec;
 use std::cell::RefCell;
 use std::rc::Rc;
 
 use crate::buffer::ChannelInterpretation;
 
-const LEN: usize = crate::BUFFER_SIZE as usize;
+/// size of a buffer, 128 samples per spec
+const BUFFER_SIZE: usize = crate::BUFFER_SIZE as usize;
```

Comment:
Not sure how I feel about 'shadowing' this const, but maybe all in all BUFFER_SIZE should just be `usize` always. I'll make a separate issue